File: inspect_otomotif.py
import sys
import pandas as pd

def main():
    try:
        path = r"C:\Users\USER\OneDrive\Desktop\Kaizen\Sample data\Otomotif\Laporan Penjualan Jan25-Jun26.xls"
        
        # Determine engine
        df = None
        for engine in [None, 'xlrd', 'openpyxl', 'pyxlsb']:
            try:
                df = pd.read_excel(path, nrows=5, engine=engine)
                print(f"Success with engine: {engine}")
                break
            except Exception as e:
                print(f"Failed with {engine}: {e}")
        
        if df is None:
            print("Could not read file.")
            return

        print("\nColumns:")
        for col in df.columns:
            print(f"- {col}")
        
        print("\nFirst row:")
        print(df.iloc[0].to_dict())
        
    except Exception as e:
        print("Fatal error:", e)

if __name__ == "__main__":
    main()

File: inspect_otomotif.mjs
import xlsx from "xlsx";
import fs from "fs";

try {
  const path = "C:\\Users\\USER\\OneDrive\\Desktop\\Kaizen\\Sample data\\Otomotif\\Laporan Penjualan Jan25-Jun26.xls";
  const workbook = xlsx.readFile(path);
  const sheetName = workbook.SheetNames[0];
  const worksheet = workbook.Sheets[sheetName];
  const data = xlsx.utils.sheet_to_json(worksheet, { header: 1 });
  
  for(let i=3; i<15; i++) {
    console.log(`Row ${i}:`, data[i]);
  }
} catch (e) {
  console.error("Error:", e.message);
}
